lib/scraper.js

```diff
@@ -53,46 +78,81 @@ Scraper.prototype.getResourceHandler = function getHandler (resource) {
 	var depth = resource.getDepth();
 	var depthGreaterThanMax = self.options.maxDepth && depth >= self.options.maxDepth;
 
-	switch (true) {
-		case depthGreaterThanMax: return _.noop;
-		case resource.isCss(): return loadCss;
-		case resource.isHtml(): return loadHtmlAndCss;
-		default: return _.noop;
+	switch(true) {
+		case depthGreaterThanMax:
+			return _.noop;
+		case resource.isCss():
+			return loadCss;
+		case resource.isHtml():
+			return loadHtmlAndCss;
+		default:
+			return _.noop;
 	}
 };
 
 Scraper.prototype.loadResource = function loadResource (resource) {
 	var self = this;
+	var url = resource.getUrl();
 
-	if(!self.options.urlFilter(resource.url)){
+	if(!self.options.urlFilter(url)) {
```

Comment:
As I can see it filters urls here, then calls `requestResource` which also contains `if(!self.options.urlFilter(url))`. Seems one of them is redundant